sunpy/physics/differential_rotation.py

```diff
@@ -159,3 +161,143 @@ def solar_rotate_coordinate(coordinate,
+    rotated_time = smap.date - timedelta(seconds=dt.to(u.s).value)
+
+    # Calculate the hpc coords
+    x = np.arange(0, smap.dimensions.x.value)
+    y = np.arange(0, smap.dimensions.y.value)
+    xx, yy = np.meshgrid(x, y)
+    # the xy input array would have the following shape
+    # xy = np.dstack([xx.T.flat, yy.T.flat])[0]
+
+    # We start by converting the pixel to world
+    with warnings.catch_warnings():
+        warnings.simplefilter('ignore')
+        hpc_coords = smap.pixel_to_world(xx * u.pix, yy * u.pix)
+
+        # then diff-rotate the hpc coordinates to the desired time
+        rotated_coord = solar_rotate_coordinate(hpc_coords, rotated_time)
+
+        # To find the values that are behind the sun we need to convert them
+        # to HeliographicStonyhurst
+        findOccult = rotated_coord.transform_to(HeliographicStonyhurst)
+
+        with np.errstate(invalid='ignore'):
+            # and find which ones are outside the [-90, 90] range.
+            occult = np.logical_or(np.less(findOccult.lon, -90 * u.deg),
+                                   np.greater(findOccult.lon, 90 * u.deg))
+
+        # NaN-ing values that move to the other side of the sun
+        rotated_coord.data.lon[occult] = np.nan * u.deg
+        rotated_coord.data.lat[occult] = np.nan * u.deg
+        rotated_coord.cache.clear()
+
+        # Go back to pixel co-ordinates
+        x2, y2 = smap.world_to_pixel(rotated_coord)
+
+    # Re-stack the data to make it correct output form
+    xy2 = np.dstack([x2.T.value.flat, y2.T.value.flat])[0]
+    # Returned a masked array with the non-finite entries masked.
+    xy2 = np.ma.array(xy2, mask=np.isnan(xy2))
+    return xy2
+
+
+@u.quantity_input(dt=u.s)
+def diffrot_map(smap, dt, pad=False):
+    """
+    Function to apply solar differential rotation to a sunpy map.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map`
+        Original map that we want to transform.
+    dt : `~astropy.units.Quantity`
+        Desired interval between the input map and returned map.
+    pad : `bool`
+        Whether to create a padded map for submaps to don't loose data
+
+    Returns
+    -------
+   `~sunpy.map`
+        A map with the result of applying solar differential rotation to the
+        input map.
+    """
+    new_time = smap.date + timedelta(seconds=dt.to(u.s).value)
+
+    # Check for masked maps
+    if smap.mask is not None:
+        smap_data = np.ma.array(smap.data, mask=smap.mask)
+    else:
+        smap_data = smap.data
+
+    submap = False
+    # Check whether the input is a submap
+    if ((2 * smap.rsun_obs > smap.top_right_coord.Tx - smap.bottom_left_coord.Tx) or
+        (2 * smap.rsun_obs > smap.top_right_coord.Ty - smap.bottom_left_coord.Ty)):
+        submap = True
+        if pad:
+            # Calculating the largest distance between the corners and their rotation values
+            deltax = deltay = 0
+            for corner in product(*product([0 * u.pix], [*smap.dimensions])):
```

Comment:
why do you need the `[*`? `smap.dimensions` is a tuple, so it's already iterable.